### PR TITLE
input from crowd in testable state

### PR DESCRIPTION
**Pull request by hrybrn**

### CODE REVIEW

File: comms/CrowdInput.py [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-def standardAddress():
-    return "http://10.9.174.43/data"
-
 class CrowdInput:    
     def __init__(self, address):
         self.address = address
 
+    @staticmethod
+    def standardAddress():
+        return "http://10.9.174.43/data"
```
Comment (mildlyincompetent) [inline comms/CrowdInput.py]: Magic IP address, very bad, please use variable, thank.

File: comms/CrowdInput.py [hunk 2]
```diff
@@ -17,6 +17,7 @@ def popular(self, seconds):
         data = self.__getData__(seconds)
         move = dict()
         for player, commands in data.items():
+            move[player] = dict()
             maxCommand = None
             maxAmount = -1
```
Comment (mildlyincompetent) [inline comms/CrowdInput.py]: Magic -1, please use variable or explain in comment, thank.

File: main.py [hunk 3]
```diff
@@ -2,8 +2,17 @@
 from display.constants import *
 from display.characters import ZUCC, Human
 from display.background import Background
+from comms.CrowdInput import CrowdInput
 
 pygame.init()
+
```
Comment (trickeydan) [inline main.py]: This conflicts with `master`

File: comms/CrowdInput.py [hunk 4]
```diff
@@ -25,26 +26,30 @@ def popular(self, seconds):
...
 
             if maxAmount == 0:
-                move[player] = "none"
-            else:
-                move[player] = maxCommand
+                move[player] = 0
+            elif maxCommand == 'up':
+                move[player] = -3
+            elif maxCommand == 'down':
+                move[player] = 3
```
Comment (mildlyincompetent) [inline comms/CrowdInput.py]: y 3?

File: comms/CrowdInput.py (hunk 1 above)
Comment (trickeydan) [inline comms/CrowdInput.py]: This makes me cry slightly. Maybe move to the constants file

File: main.py [hunk 5]
```diff
@@ -2,8 +2,17 @@
...
+screen_width = 1920
```
Comment (mildlyincompetent) [inline main.py]: Constants live in constants file. Wasps live in nests constructed from paper that they produce by chewing plant matter.

File: main.py [hunk 6]
```diff
@@ -2,8 +2,17 @@
...
+screen_height = 1080
+game_title = "Soton Analytica"
+
+input_valid_time_seconds = 3
+input_type = "democracy"
```
Comment (mildlyincompetent) [inline main.py]: Soton Analytica is incompatible with democracy.

File: main.py [hunk 7]
```diff
@@ -21,21 +30,25 @@
...
         elif event.type == pygame.KEYDOWN:
-            if event.key == pygame.K_UP:
-                ZUCC.ySpeed = -3
-            elif event.key == pygame.K_DOWN:
+            if event.key == pygame.K_DOWN:
                 ZUCC.ySpeed = 3
             elif event.key == pygame.K_e:
                 ZUCC.evolve()
-        elif event.type == pygame.KEYUP:
-            ZUCC.ySpeed = 0
```
Comment (mildlyincompetent) [inline main.py]: Removal of mentions of ZUCC is discouraged. Please increase quantity of ZUCC.

Comment (trickeydan) [reply main.py]: :pineapple:

File: comms/CrowdInput.py (hunk 4 above)
Comment (hrybrn) [reply comms/CrowdInput.py]: previous setting

**Pull request merged**